pepper-apis/docs/specification/src/endpoints/user.studies.activity.uploads.yml

```diff
@@ -0,0 +1,106 @@
+parameters:
+  - $ref: '../pepper.yml#/components/parameters/studyGuid'
+  - $ref: '../pepper.yml#/components/parameters/userGuid'
+  - $ref: '../pepper.yml#/components/parameters/instanceGuid'
+
+post:
+  operationId: postActivityUploads
+  tags:
+    - Activities
+  summary: create an authorized file upload
+  description: |
+    Request for a new URL for file upload. Link is accessible for limited
+    period of time (e.g. 5 mins). Resumable uploads are supported. With
+    information in response body, client will be able to upload the file
+    directly to a Google Bucket. If resumable=true is specified in request,
+    client will assume responsibility for creating the session URI with
+    provided URL in addition to follow up requests that contain the file
+    contents (see [documentation][doc]).
+
+    **Study Admins**
+
+    Activity file upload is not supported if activity instance/question is
+    read-only or otherwise inaccessible, unless operator is a study admin.
+
+    **Temporary Users**
+
+    This API endpoint does not allow access by temporary users.
+
+    [doc]: https://cloud.google.com/storage/docs/performing-resumable-uploads
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          type: object
+          required:
+            - questionStableId
+            - fileName
+            - fileSize
+            - resumable
+          properties:
+            questionStableId:
+              type: string
+              description: The question identifier
+            fileName:
+              type: string
+              maxLength: 255
```

Comment:
What's going to happen if the length of the string is more than 255 bytes? Truncation, failed request, ...?

Reply:
Request will be rejected with a `400` status code. See updated description of `400` response.

Reply:
That feels like a weird weird situation. Right now it seems like we have multiple disparate status codes for bad parameters. A file size too large is returning a `422` while an overly long string returns a `400`. Could we collapse these into a single error (`400` with a `BAD_PARAMETER` code, and explaining the error in the message)?

Reply:
You're right, a 400 does seem to be a better fit! I'll switch this over to a `400` with the usual `BAD_PAYLOAD` error code.

Reply:
Sounds good! It looked like the included message clearly indicates the error, so I think this is set after the change.

Reply:
Done! d52f3cabfa9bfd37262aea867e0b9f0f26a06dc0